platform/wayland/cog-platform-wl.c

```diff
@@ -568,44 +565,64 @@ pointer_on_enter(void              *data,
                  wl_fixed_t         fixed_x,
                  wl_fixed_t         fixed_y)
 {
-    CogWlPlatform *platform = data;
-    CogWlDisplay  *display = platform->display;
+    CogWlSeat *seat = data;
 
-    display->seat.keyboard.serial = serial;
-    display->seat.pointer.serial = serial;
-    display->seat.pointer.surface = surface;
+    if (pointer != seat->pointer_obj) {
+        g_critical("%s: Got pointer %p, expected %p.", G_STRFUNC, pointer, seat->pointer_obj);
+        return;
+    }
+
+    seat->display->seat_default = seat;
+
+    CogWlPlatform *platform =
+        wl_surface_get_user_data(surface); // Platform is attached the context of the wl_surface. Check create_window();
```

Comment:
This is the way that I did to attach the Platform as a object associated to the surface. Check this: https://github.com/Igalia/cog/pull/623/commits/a321b21181571ea271b3465d181e2581e9bc38d7#diff-7342248ee4ad822cd294d9c8341a52fd10b3d1df4dd2a202ccb926df67d51034R133.